catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get/mod.rs

```diff
@@ -107,19 +103,10 @@ pub(crate) async fn endpoint(
 }
 
 /// Returns a Catalyst ID for the given stake address.
+#[allow(clippy::unused_async)]
 async fn catalyst_id_from_stake(
-    session: &CassandraSession, address: StakeAddress,
+    _session: &CassandraSession, _address: StakeAddress,
 ) -> anyhow::Result<Option<CatalystId>> {
-    let mut result: Vec<_> = Query::execute(session, QueryParams {
-        stake_address: address.into(),
-    })
-    .and_then(|r| r.try_collect().map_err(Into::into))
-    .await
-    .map(|mut res: Vec<_>| {
-        res.sort_by(|a, b| a.slot_no.cmp(&b.slot_no));
-        res
-    })
-    .context("Failed to query Catalyst ID from stake address")?;
-
-    Ok(result.pop().map(|v| v.catalyst_id.into()))
+    // TODO: This function should be entirely removed as a part of https://github.com/input-output-hk/catalyst-voices/issues/2532
```

Comment:
Why we cannot remove this function already ?
Am I correct that with this change RBAC endpoint will return 404 every time if stake address is passed as lookup argument ?

Reply:
Removing this function will require to remove more code from the `catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get` module. I prefer not to change the unrelated code that will be removed anyway as a part of another task (https://github.com/input-output-hk/catalyst-voices/issues/2532).

Reply:
> Am I correct that with this change RBAC endpoint will return 404 every time if stake address is passed as lookup argument ?

Yes.